rgf/utils.py

```diff
@@ -500,16 +536,19 @@ def fit(self, X, y, sample_weight=None):
             self._is_sparse_train_X = False
         self._n_samples, self._n_features = X.shape
         sample_weight = self._get_sample_weight(sample_weight)
-        check_consistent_length(X, y, sample_weight)
-        check_classification_targets(y)
-
-        self._classes = sorted(np.unique(y))
-        self._n_classes = len(self._classes)
+        if self.is_classification:
```

Comment:
Let me remind you your own comment about branching based on the task-/RGF-type condition: https://github.com/RGF-team/rgf_python/pull/106#issuecomment-351068351.